components/base/sensorcontrolled/movestraight.go

```diff
@@ -54,6 +56,15 @@ func (sb *sensorBase) MoveStraight(
 		}
 	}
 
+	// if loop is tuning, return an error
+	// if loop has been tuned but the values haven't been added to the config, error with tuned values
+	if sb.loop != nil && sb.loop.GetTuning(ctx) {
+		return control.TuningInProgressErr(sb.Name().ShortName())
+	} else if (sb.configPIDVals[0].NeedsAutoTuning() && !(*sb.tunedVals)[0].NeedsAutoTuning()) ||
+		(sb.configPIDVals[1].NeedsAutoTuning() && !(*sb.tunedVals)[1].NeedsAutoTuning()) {
+		return control.TunedPIDErr(sb.Name().ShortName(), *sb.tunedVals)
+	}
+
```

Comment:
let's make this a bit more readable. You can:
1 - use a case structure.
2—Write an in-line, anonymous function that checks if the PID values need Autotuning and passes in the four PID values for your second case.

Reply:
tbh i think these should just be two separate if statements with their own comments. Don't see a reason for them to be grouped together and I personally find it weird when switch statements have a bunch of logic for cases(fine if peeps disagree tho). Agree with the helper function, its something I wanted to add as part of the refactors